src/charm.py

```diff
@@ -241,6 +244,7 @@ def _on_config_changed(self, _):
         self._configure()
 
     def _on_upgrade_charm(self, _):
+        self._stored.is_upgrade = True
```

Comment:
If we don't reset this stored value to False on stop, then its name `is_upgrade` is misleading, because what it really tells you is "has undergone an upgrade in the past".

Would it make sense to call it `_stored.fresh_install` instead?